src/core/border_router/routing_manager.cpp

```diff
@@ -636,7 +636,9 @@ void RoutingManager::StartRouterSolicitationDelay(void)
 {
     uint32_t randomDelay;
 
-    VerifyOrExit(!mRouterSolicitTimer.IsRunning() && mRouterSolicitCount == 0);
+    VerifyOrExit(!IsRouterSolicitationInProgress());
+
+    OT_ASSERT(mRouterSolicitCount == 0);
```

Comment:
```suggestion
```

Reply:
may still fine to keep the assert?

Reply:
`IsRouterSolicitationInProgress` is already making this for sure.
If we want to add the `assert`, better to add after every usage of `IsRouterSolicitationInProgress` but I suggest don't do that - it's meaningless :)

Reply:
> `IsRouterSolicitationInProgress` is already making this for sure.

No, it isn't. 

`RouterSolicitationInProgress` means `mRouterSolicitTimer.IsRunning() || mRouterSolicitCount > 0`.

So, it's still possible that `mRouterSolicitCount > 0`. But here `mRouterSolicitCount` must be 0, so it's an additional condition.